File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Array_Sorting/Longest_Consecutive_Sequence/longest_consecutive_sequence.ts
// Longest Consecutive Sequence - TypeScript implementation
// Time Complexity: O(n) where n is the length of nums
// Space Complexity: O(n)

export function longestConsecutive(nums: number[]): number {
  if (nums.length === 0) return 0;

  const numSet = new Set(nums);
  let maxLength = 0;

  for (const num of numSet) {
    // Only start counting if this is the beginning of a sequence
    // (i.e., num - 1 is not in the set)
    if (!numSet.has(num - 1)) {
      let currentNum = num;
      let currentLength = 1;

      // Count consecutive numbers
      while (numSet.has(currentNum + 1)) {
        currentNum++;
        currentLength++;
      }

      maxLength = Math.max(maxLength, currentLength);
    }
  }

  return maxLength;
}

File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Array_Sorting/Longest_Consecutive_Sequence/longest_consecutive_sequence.cs
// Longest Consecutive Sequence - C# implementation
// Time Complexity: O(n) where n is the length of nums
// Space Complexity: O(n)

public class Solution
{
    public int LongestConsecutive(int[] nums)
    {
        if (nums == null || nums.Length == 0)
        {
            return 0;
        }

        var numSet = new HashSet<int>(nums);
        int maxLength = 0;

        foreach (int num in numSet)
        {
            // Only start counting if this is the beginning of a sequence
            // (i.e., num - 1 is not in the set)
            if (!numSet.Contains(num - 1))
            {
                int currentNum = num;
                int currentLength = 1;

                // Count consecutive numbers
                while (numSet.Contains(currentNum + 1))
                {
                    currentNum++;
                    currentLength++;
                }

                maxLength = Math.Max(maxLength, currentLength);
            }
        }

        return maxLength;
    }
}

